File: DemoMethods/Basic/MultiMapIndexingQuery.cs
using System;
using System.Linq;
using System.Web.Http;
using DemoMethods.Helpers;
using DemoMethods.Indexes;
using Raven.Client;

namespace DemoMethods.Basic
{
    public partial class BasicController : DemoApiController
    {
        [HttpGet]
        [Demo("MultiMap Indexing Query", DemoOutputType.Flatten, demoOrder: 100)]
        public object MultiMapIndexingQuery(string country = "USA")
        {
            using (var session = DocumentStoreHolder.Store.OpenSession())
            {
                return session.Query<NameAndCountry.Result, NameAndCountry>()
                    .Search(x => x.Country, country)
                    .Select(x => new {x.Name, x.Id})
                    .ToList();
            }
        }
    }
}

File: DemoServer/PythonDemo/demo_methods/Basic/MultiMapIndexingQuery.py
from demo.holder import DocumentStoreHolder
from demo.indexes import NameAndCountry


class MultiMapIndexingQuery(object):
    @staticmethod
    def multi_map_indexing_query(country="USA"):
        with DocumentStoreHolder.get_store().open_session() as session:
            return [{"Id": result.Id, "Name": result.Name} for result in
                    list(session.query(index_name=NameAndCountry.__name__, object_type=NameAndCountry.Result,
                                       wait_for_non_stale_results=True).
                         search("Country", country).
                         select("Name", "Id"))]
